### PR TITLE
Fixes a reference cycle issue with `WrapMany(err ...error)`

### PR DESCRIPTION
**Pull request by wcharczuk**
If you Wrapped the same exception multiple times in `WrapMany` it would create a reference/pointer cycle and horrendously explode calls to .Error(). Fixes issue and adds regression test.

### CODE REVIEW

File: exception_test.go [hunk 1]
```diff
@@ -113,5 +124,5 @@ func TestWrapManyNil(t *testing.T) {
 
 	err := WrapMany(ex1, ex2, ex3)
 	a.Nil(err)
-	a.Equal(nil, err) //UGH.
+	a.Equal(nil, err)
```
Comment (John-Steidley) [inline exception_test.go]: Why is this necessary after a.Nil(err)?

Comment (wcharczuk) [inline exception_test.go]: because (*Type)(nil) is not == to <nil>

File: exception_test.go [hunk 2]
```diff
@@ -102,6 +102,17 @@ func TestWrapMany(t *testing.T) {
...
+}
+
+func TestWrapManyWithCycle(t *testing.T) {
+	a := assert.New(t)
+
+	ex1 := errors.New("This is an error")
+	err := WrapMany(ex1, ex1)
+
+	a.NotNil(err)
+	a.NotEmpty(err.Error())
```
Comment (christopheralme) [inline exception_test.go]: Can you check the content of the error for the wrapped case? It'll be cool to be explicit for the behavior in the test/understanding what's going on

**Comment by christopheralme** [comment]
lgtm contingent response to test comments

File: exception_test.go (hunk 1 above)
Comment (John-Steidley) [inline exception_test.go]: I thought it did: https://play.golang.org/p/a5yxcn3GKT

What's the difference between my code and this?

Comment (wcharczuk) [inline exception_test.go]: https://play.golang.org/p/EoNOAVsWrL
